### PR TITLE
Feature/visualization

### PR DESCRIPTION
**Pull request by carcasonne**
* Add the front end
* Add new API for getting geojson for a road section

### CODE REVIEW

File: traffic_speed_prediction/frontend/templates/frontend/finmap.html [hunk 1]
```diff
@@ -7,48 +7,234 @@
...
+
+                //get the full coordinates of this road
+                //const newPromise = fetch_coordinates(closestRoad)
+                const geodataPromise = fetch_geodata(closestRoad, closestRoadSection)
+
+                geodataPromise.then(function(geodata) {
+                    // Bit of a tricky path, just the way the json is formatted
+                    //const finalCoords = newResult.features[0].geometry.coordinates//JSON.stringify(newResult.features)
+
+                    //newResult.features.forEach(element => {
```
Comment (DenizIsik58) [inline traffic_speed_prediction/frontend/templates/frontend/finmap.html:65]: Check if comments needs to be removed

File: traffic_speed_prediction/frontend/templates/frontend/finmap.html [hunk 2]
```diff
@@ -7,48 +7,234 @@
...
+
+                    load_road_from_geojson(predition, numberOfRoads.toString(), numberOfRoads.toString(), geodata)
+                    numberOfRoads++;
+
+                    if(debug) 
+                    {
+                        console.log("THE ROAD WE ARE LOOKING FOR HAS ID: " + result.toadId)
+                        console.log("Coords at mouse: " + mouseCoords);
+                        console.log("closestRoad: " + closestRoad)
+                        //console.log("Coords of road: " + newResult.features[0].geometry.coordinates)
```
Comment (DenizIsik58) [inline traffic_speed_prediction/frontend/templates/frontend/finmap.html:90]: Maybe clean console.log

File: traffic_speed_prediction/util/db/database_commands.py [hunk 3]
```diff
@@ -89,24 +91,54 @@ def getNearestCoordsAndPredictions(lat, lon):
                 road_sect.append(int(str(road_section.weatherSymbol)[1:]))
                 road_sect.append(int(road_section.roadMaintenanceClass))
                 road_sect.append(float((road_section.freeFlowSpeed1)))
-                #print(road_section.road.Road_number)
+                print(road_section.road.Road_number)
         print(road_sect)
         print("NEAREST: " + str(nearest_distance))
```
Comment (DenizIsik58) [inline traffic_speed_prediction/util/db/database_commands.py:96]: maybe remove print statements

File: traffic_speed_prediction/util/db/database_commands.py [hunk 4]
```diff
@@ -89,24 +91,54 @@ def getNearestCoordsAndPredictions(lat, lon):
...
+    def getInfoForPredictionByLatAndLon(lon2, lat2):
 
         nearest_distance = 10000
         road_sect = []
         la = 0
         lo = 0
 
+        r = 6371 # Radius of earth in kilometers. Use 3956 for miles. Determines return value units.
+
+        print("Query points:")
```
Comment (DenizIsik58) [inline traffic_speed_prediction/util/db/database_commands.py:110]: maybe remove print statements

File: traffic_speed_prediction/util/db/database_commands.py [hunk 5]
```diff
@@ -89,24 +91,54 @@ def getNearestCoordsAndPredictions(lat, lon):
...
+            deltaLon = lat1 - lat2
+            deltaLat = lon1 - lon2
+            
+            #Using euclidean is faster, but is super imprecise. always gets the same result
             temp_distance = math.sqrt(
-                math.pow(road_section.lat - lat, 2) + math.pow(road_section.lon - lon, 2))
+                deltaLon*deltaLon+ deltaLat*deltaLat
+            )
+
+            #print("recieved lat:")
```
Comment (DenizIsik58) [inline traffic_speed_prediction/util/db/database_commands.py:126]: print statements here

**Review by antbrr — APPROVED** [review]
looks good

**Pull request merged**